File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs)==0:
            return ""
        prefix=strs[0]
        for word in strs[1:]:
            if len(prefix)> len(word):
                prefix = prefix[:len(word)]
            for index in range(len(prefix)):
                if prefix[index]!=word[index]:
                    prefix = prefix[0:index]
                    break
            if len(prefix) == 0:
                return ""
        return prefix
        
       

File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) {
            return "";
        }
        std::string prefix = strs[0];
        for (std::string word: strs) {
            if (prefix.length() > word.length()) {
                prefix = prefix.substr(0, word.length());
            }
            for (int index = 0; index < prefix.length(); index++) {
                if (prefix[index] != word[index]) {
                    prefix = prefix.substr(0, index);
                    break;
                }
            }
            if (prefix.length() == 0) {
                return "";
            }
        }
        return prefix;
    }
};